src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -1519,8 +1519,10 @@ class OpenShiftDSL implements Serializable {
         public <V> V withEach(Closure<V> body) {
             defaultProjectsForObjectList();
             List<String> names = names();
+            if (names == null || names.size() == 0)
+                return;
             List<String> projects = projects();
-            if (projects == null) {
+            if (projects == null || projects.size != names.size()) {
```

Comment:
should this be projects.size() ?

Reply:
ahh ... good catch ... not so safe after all :-)

Reply:
and another instance of stupid groovy ;-)

Reply:
Is there any kind of groovy linter that we could add to the testing for these repos?